frontend/src/containers/Ledger.js

```diff
@@ -153,34 +153,41 @@ export class Ledger extends Component {
 
   }
 
+  loadCollectiveData() {
+    const { collective, params, fetchProfile } = this.props;
+    if (collective && collective.id) return Promise.resolve(collective);
+    return fetchProfile(params.slug).then(result => {
+      return result.groups[result.slug];
+    });
+  }
+
   componentWillMount() {
     const { 
-      collective,
       fetchUsers,
       fetchPendingExpenses,
       fetchTransactions,
-      loadData,
       params
     } = this.props;
 
-    switch (params.action) {
-      case 'approve':
-        this.approveExp(params.expenseid);
-        break;
-      case 'reject':
-        this.rejectExp(params.expenseid);
-        break;
-    }
-
-    if (loadData) { // useful when not server-side rendered
-      fetchProfile(collective.slug);
-    }
-    Promise.all([
-      fetchUsers(collective.slug),
-      fetchPendingExpenses(collective.slug),
-      fetchTransactions(collective.slug)
-      ])
-    .then(() => scrollToExpense());
+    this.loadCollectiveData()
+      .then((collective) => {
+        this.collective = collective;
```

Comment:
I think it's better to use reducers. If the user goes back to the collective page, we'll have to refetch the collective data.